### PR TITLE
Fix bogus uses of BIO_sock_should_retry

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Hi this applies to all branches (except the hunk in do_server which is not needed for 1.0.2).

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
Note that openssl s_time command actually fails after a few seconds because it quickly runs
out of sockets because all 65535 possible port numbers are in FIN WAIT2.

I know how to fix that, but it will require to use SO_LINGER  and set the linger timeout
to zero.  Do you see problems with that idea?

File: apps/s_time.c [hunk 1]
```diff
@@ -403,7 +403,8 @@ static SSL *doConnection(SSL *scon, const char *host, SSL_CTX *ctx)
     /* ok, lets connect */
     for (;;) {
         i = SSL_connect(serverCon);
-        if (BIO_sock_should_retry(i)) {
+        if (SSL_get_error(serverCon, i) == SSL_ERROR_WANT_CONNECT) {
+abort();
```
Comment (dot-asm) [inline apps/s_time.c]: ???

Comment (bernd-edlinger) [inline apps/s_time.c]: pah... yes, debug-code I forgot to remove!!!

Comment (bernd-edlinger) [inline apps/s_time.c]: actually I begin to wonder of all the want_connect and the want_read / write stuff
really is good for anything, when there is no way ho to enable non-blocking ops.

Comment (bernd-edlinger) [inline apps/s_time.c]: fixed.  but I think maybe adding a -nbio option would be nice.

**Comment by bernd-edlinger** [comment]
Hmm....
Actually maybe we should better remove all the unreachable non-blocking stuff
from s_time, effectively reverting the PR #3925 and make sure that the test case
does actually succeed in the blocking way

**Comment by dot-asm** [comment]
As for SO_LINGER question. Yes, I'd say it's perfectly appropriate. Here or elsewhere?

On related side note. master (haven't checked 1.1) s_time didn't do anything for me. Well, it exited with non-zero code, but didn't provide any error message. As it turned out I mistyped cipher name. It should have said that. E.g. 1.0.2 does say it. Well, it's not optimal in 1.0.2 either, it says "no cipher", then says "collecting connection statistics, gets another "no cipher" and only then exits. It could have exited at first "no cipher"...

**Comment by bernd-edlinger** [comment]
So I removed the s_time.c completely for this PR and will create another PR for the cleanup
of s_time.c

**Comment by bernd-edlinger** [comment]
yes. that happened to me too....

**Comment by bernd-edlinger** [comment]
regarding SO_LINGER I hope all O/S support it.
I would inject SO_LINGER code before the close socket.
send RST packet instead of FIN viola.

**Comment by bernd-edlinger** [comment]
actually in master it complained about the cipher, if 1.0.2 is so quiet that is another bug.

**Comment by dot-asm** [comment]
> actually in master it complained about the cipher, if 1.0.2 is so quiet that is another bug.

For me it's exactly opposite. I've double-checked.

**Comment by bernd-edlinger** [comment]
maybe I mixed that up,  thanks!

**Comment by bernd-edlinger** [comment]
I assume approval is for 1.1.0 & master...

**Comment by dot-asm** [comment]
> I assume approval is for 1.1.0 & master...

I normally operate under assumption "no label -> master only". But declaring in text works too. It was declared, so yes.

**Comment by bernd-edlinger** [comment]
ok, I will try to set the labels in future.

**Comment by bernd-edlinger** [comment]
Merged to master as daaaa3c.
Merged to 1.1.0 as 56c13d2.
Thanks!

**Pull request closed without merging**